### ISSUE TITLE
Unsupervised learning using keras

### ISSUE THREAD

**Issue by vinayakumarr** [fchollet/keras#3464]
Has any body tried to do unsupervised learning using keras. Could you please help me..i am new to keras

**Comment by marcj** [comment]
> Has any body tried to do unsupervised learning using keras.

Yes. Is this your only question?

**Comment by vinayakumarr** [comment]
Yes....unsupervised learning in deep learning using keras. I know auto encoder is unsupervised...but is there any deep learning algorithm for unsupervised learning

**Comment by farizrahman4u** [comment]
You will have to be a lot more specific.

**Comment by vinayakumarr** [comment]
I have data which is of 9 class label and 1800 features. This is basically a classification. I want to apply unsupervised learning. Could you please suggest us regarding this and also is there ant basic code to do this.

**Comment by farizrahman4u** [comment]
If you have labels then why unsupervised?

**Comment by vinayakumarr** [comment]
I would like to try unsupervised classification. To try out new things

**Comment by swayson** [comment]
Never tried it myself, but perhaps this paper could be of use: 

NEURAL NETWORK-BASED CLUSTERING USING PAIRWISE
CONSTRAINTS

http://arxiv.org/pdf/1511.06321v5.pdf

**Comment by abishekk92** [comment]
Given the context, I am not sure what exactly you're looking to achieve, that being said, you can take look at [how to implement an autoencoder using Keras](https://blog.keras.io/building-autoencoders-in-keras.html)

**Comment by gaw89** [comment]
@swayson I have been trying to implement the method in that paper the past couple of weeks, but for the life of my I can't make it work.  Do you happen to know if anyone has tried it/had any success with it yet?

**Comment by rjpg** [comment]
I saw the link : https://blog.keras.io/building-autoencoders-in-keras.html

Is well done BUT is missing something critical :

How to apply autoencoders in classification using keras ? 

Like , extend the examples to : given on image of a digit it tells you what is the digit it self (?) 

For this we need to know how (after autoencoder unsupervised training):

1-to slice the NN in half on the last encoder (before the decode starts)
2- freeze the the wights of the encoders 
3- add or concat  a new NN in front of the last encoding layer 
4- (post) train this concat NN with a softmax output layer to classify the digit ...

Example with all this would be super

**Comment by jolespin** [comment]
Here I give an example of using an autoencoder on the iris dataset.  You can use that embedding w/ a clustering algorithm in scikit-learn.  I have been experiencing an issue where one of the dimensions gets zeroed out :/ 

https://stackoverflow.com/questions/47297526/how-to-prevent-simple-keras-autoencoder-from-over-compressing-data
